.github/workflows/unit_tests.yaml

```diff
@@ -53,6 +53,10 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
+    - name: Check for typos in docs
+      uses: crate-ci/typos@592b36d23c62cb378f6097a292bc902ee73f93ef # version 1.0.4
+      with: 
+        files: ./lib/spack/docs/ ./lib/spack/spack/*.py
```

Comment:
I can't find it in the `crate-ci` docs -- does this just get the py files in lib/spack/spack, or is it recursive?  We also should include `lib/spack/llnl` and `var/spack/repos/builtin/packages/*/*.py`.

Reply:
@tgamblin I added it to the maximum number of places that would have maximum benefit without making it harder for the developer. For example, if we add to lib/spack/spack, while it does find many spelling errors, it also targets variable names, e.g.,:

```python
error: `fullset` should be `fullest`
  --> lib/spack/spack/util/environment.py:1041:25
     |
1041 |     def set_intersection(fullset, *args):
```
This is intentionally 'fullset` (which is not a word but implies full set) and should be renamed to fullset. When I saw errors like this, I realized that adding it to spell check code would be a huge annoyance for the developer - we already have enough style checks that are hard to deal with (pending your recent changes of course).

But for documentation? That's the user facing thing, and it denotes how professional our project is. Arguably spelling mistakes in code won't be seen by most, just developers. But documentation will. So this explains my choice to add spell checking to the documentation but not to the code.

There could be some intermediate to just spell check (and show) errors in the code and then it's up to the developer to fix if he/she wants, but it wouldn't result in an error code. My suggestion thus is the following:

1. to get the action in with this current setup to handle the docs, this is priority
2. to have two different PRs for each of adding spell checking to packages and the code, each with possibly a different strategy depending on what is found (small number of cases we can add exceptions and let the check run, larger number of cases we can fix and have run but not make it fail a test).

Reply:
> When I saw errors like this, I realized that adding it to spell check code would be a huge annoyance for the developer

I'm assuming the number of cases is high enough and growth is high enough to not be worth adding these to `typos.toml` file as "approved" spellings?

Reply:
We could, but every time someone uses a variable that looks like a misspelled word it’s going to trigger an error.